test/gurka/test_ramps_tc.cc

```diff
@@ -279,3 +279,148 @@ TEST(RampsTCs, test_tc_infer) {
       result, maneuver_index, "Turn left onto KJ.", "Turn left onto KJ.",
       "Turn left onto KJ. Then You will arrive at your destination.", "Continue for 50 meters.");
 }
+
+namespace {
+void check_edge_classification(baldr::GraphReader& graph_reader,
+                               const gurka::nodelayout& nodes,
+                               const std::string& b,
+                               const std::string& e,
+                               baldr::RoadClass rc) {
+  const auto edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, nodes, b, e));
+  EXPECT_EQ(edge->classification(), rc);
+}
+} // namespace
+
+TEST(LinkReclassification, test_use_refs) {
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+  A------------B-------------C-----------D
+                \     G     /
+                 \    |    K----------L
+     I------------J   |   /
+                   \  |  /
+                    \ | /
+                     \|/
+                      F
+                      |
+                      |
+                      E
+  )";
```

Comment:
this diagram wins the award for **best ascii art in gurka test** :trophy:

Reply:
haha :joy: Tank you!))